### DISCUSSION TITLE
when close cef browser cleare local storage

### DISCUSSION THREAD

**Discussion by AbhiDevelop** [cefsharp/CefSharp#4169] [Q&A]
when close CEF sharp browser cleared local storage in window application then what i do please suggest any one to me

**Comment by amaitland** [comment]
Your options are something like the following
- Use an in memory cache (don't specify a `CachePath`) so nothing is persisted to disk
- Execute javascript to clear storage (must be done before the browser is closed). https://developer.mozilla.org/en-US/docs/Web/API/Storage/clear
- Use DevTools protocol to clear storage for a domain (must be done before browser is closed). There are a number of methods that might be suitable, you'll need to determine which if any will work. 
  - https://chromedevtools.github.io/devtools-protocol/tot/Storage/#method-clearDataForOrigin
  - https://chromedevtools.github.io/devtools-protocol/tot/DOMStorage/#method-clear
  - https://chromedevtools.github.io/devtools-protocol/tot/Network/#method-clearBrowserCache

See https://github.com/cefsharp/CefSharp/issues/3165 for details on strongly typed `DevTools APU` that's included as part of `CefSharp`
